awx/main/models/jobs.py

```diff
@@ -1182,18 +1182,19 @@ def _accept_or_ignore_variables(self, data, errors, _exclude_errors=()):
             for key in unallowed_vars:
                 rejected[key] = data.pop(key)
 
-        if 'days' in data:
-            try:
-                if type(data['days']) is bool:
-                    raise ValueError
-                if float(data['days']) != int(data['days']):
-                    raise ValueError
-                days = int(data['days'])
-                if days < 0:
-                    raise ValueError
-            except ValueError:
-                errors_list.append(_("days must be a positive integer."))
-                rejected['days'] = data.pop('days')
+        if self.job_type in ('cleanup_jobs', 'cleanup_activitystream'):
+            if 'days' in data:
+                try:
+                    if type(data['days']) is bool:
```

Comment:
given the traceback that you had for this, maybe this would be better as `if isinstance(data['days'], (bool, type(None))):` to catch that case in the event that the client gave a weird none-type days for a valid job type